train.py

```diff
@@ -0,0 +1,132 @@
+                    .with_spectral_kurtosis()\
+                    .with_spectral_skewness()\
+                    .with_spectral_rolloff()\
+                    .with_spectral_flatness()\
+                    .with_mfcc()\
+                    .get_feature_matrix()
+                t = []
+                total_matches = 0
+                for time in onset.get_times():
+                    truth = get_truth(os.path.join(train_folder, folder), time)
+                    t.append(truth)
+                    total_matches += len(truth)
+                X.extend(f)
+                y.extend(t)
+                break
+    y = MultiLabelBinarizer(classes=classes).fit_transform(y)
+    print("Onset detection captured {0} out of {1} training data events".format(total_matches, \
+                                                                                get_total_events(train_folder)))
+    return np.array(X), y
+
+def print_report(truth, pred, folds=10):
+    print("Using cross validation with ", folds, "folds")
+    print("Label ranking average precision: ",  metrics.label_ranking_average_precision_score(truth, pred))
+    print("Coverage error", metrics.coverage_error(truth, pred))
+    print("Ranking loss", metrics.label_ranking_loss(truth, pred))
+    class_results = []
+    for i, c in enumerate(classes):
+        c_truth = np.array(truth)[:, i]
+        c_pred = np.array(pred)[:, i]
+        results = [c, "%.3f" % metrics.accuracy_score(c_truth, c_pred)]
+        if np.sum(c_pred) == 0:
+            results.append("n/a")
+        else:
+            results.append("%.3f" % metrics.precision_score(c_truth, c_pred))
+        if np.sum(c_truth) == 0:
+            results.append("n/a")
+        else:
+            results.append("%.3f" % metrics.recall_score(c_truth, c_pred))
+        class_results.append(results)
+    print(tabulate(class_results, headers=["Class", "Accuracy", "Precision", "Recall"]))
+
+def test_model(clf, X, y, folds=10):
+    skf = KFold(n_splits=folds)
+    truth = []
+    pred = []
+    for train_index, test_index in skf.split(X):
+        X_train, X_test = X[train_index], X[test_index]
+        y_train, y_test = y[train_index], y[test_index]
+        clf.fit(X_train, y_train)
+        p = clf.predict(X_test)
+        truth.extend(y_test)
+        pred.extend(p)
+    return truth, pred
+
+def train():
+    X, y = get_data()
+    clf = MLPClassifier(solver='lbfgs', alpha=1e-5, hidden_layer_sizes=(15,))
+    truth, pred = test_model(clf, X, y)
+    print_report(truth, pred)
+    print("Would you like to export this trained model? [y/n]")
+    export = input().lower().strip() == "y"
+    if export:
+        print("Enter file name")
+        file = input()
+        joblib.dump(clf, './trained_models/' + file + '.pkl')
+        print("Exported model " + file + " in trained models")
+
+
+if __name__ == '__main__':
+    train()
```

Comment:
you forgot a blank newline at the end of file -- going to have to reject this pr due to low code quality

Reply:
shit, wouldn't blame you if you dumped me as a project partner too..